### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Remove dead code - remove checks run always earlier in `src/memory_provider.c`.

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: src/provider/provider_coarse.c [hunk 1]
```diff
@@ -1643,9 +1642,7 @@ static umf_result_t coarse_memory_provider_allocation_merge(void *provider,
                                                             void *lowPtr,
                                                             void *highPtr,
                                                             size_t totalSize) {
-    if (provider == NULL || lowPtr == NULL || highPtr == NULL ||
-        ((uintptr_t)highPtr <= (uintptr_t)lowPtr) ||
-        ((uintptr_t)highPtr - (uintptr_t)lowPtr >= totalSize)) {
```
Comment (lukaszstolarczuk) [inline src/provider/provider_coarse.c:1648]: just asking, to confirm - the check is slightly different in `src/memory_provider.c`: `(uintptr_t)highPtr - (uintptr_t)lowPtr > totalSize`

File: src/provider/provider_coarse.c [hunk 2]
```diff
@@ -1643,9 +1642,7 @@ static umf_result_t coarse_memory_provider_allocation_merge(void *provider,
...
+    if (provider == NULL) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
```
Comment (lukaszstolarczuk) [inline src/provider/provider_coarse.c:1648]: can we unify this for all funcionts perhaps...?
I can see `purge_force` functions check for null ptr and we should probably have this check in the upper layer (in `src/memory_provider.c`). The same e.g. in `ipc_handle_size` functions - all of them individually check if `size == null`... Does this make sense...?

**Comment by bratpiorka** [comment]
please rebase

File: src/provider/provider_coarse.c (hunk 1 above)
Comment (ldorau) [reply src/provider/provider_coarse.c:1583]: The check fixed in `src/memory_provider.c`
Done.

File: src/provider/provider_coarse.c (hunk 2 above)
Comment (ldorau) [reply src/provider/provider_coarse.c]: Done

**Comment by ldorau** [comment]
> please rebase

Done

File: src/provider/provider_coarse.c (hunk 1 above)
Comment (ldorau) [reply src/provider/provider_coarse.c:1583]: @lukaszstolarczuk ping

File: src/provider/provider_coarse.c (hunk 2 above)
Comment (ldorau) [reply src/provider/provider_coarse.c]: @lukaszstolarczuk ping

**Review by bratpiorka — APPROVED** [review]

File: src/provider/provider_coarse.c (hunk 1 above)
Comment (ldorau) [reply src/provider/provider_coarse.c:1583]: @lukaszstolarczuk ping

File: src/provider/provider_coarse.c (hunk 2 above)
Comment (ldorau) [reply src/provider/provider_coarse.c]: @lukaszstolarczuk ping

Comment (lukaszstolarczuk) [reply src/provider/provider_coarse.c]: thx!

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**